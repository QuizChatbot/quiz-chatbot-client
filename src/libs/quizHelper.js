const getBlankQuest = () => ({
  subject: '',
  category: '',
  question: '',
  choice_0: '',
  choice_1: '',
  choice_2: '',
  open: true
})

const getQuestFromProps = quest => {
  if (quest === undefined) return {}
  const { subject, category, question, choices } = quest
  return {
    subject,
    category,
    question,
    choice_0: choices[0],
    choice_1: choices[1],
    choice_2: choices[2]
  }
}

const getQuizStatefromQuest = ({
  subject = '',
  category = '',
  question = '',
  choice_0 = '',
  choice_1 = '',
  choice_2 = ''
}) => ({
  subject,
  category,
  question,
  choice_0,
  choice_1,
  choice_2,
  isEditing: {
    subject: false,
    category: false,
    question: false,
    choice_0: false,
    choice_1: false,
    choice_2: false
  },
  open: false
})

export { getBlankQuest, getQuestFromProps, getQuizStatefromQuest }
