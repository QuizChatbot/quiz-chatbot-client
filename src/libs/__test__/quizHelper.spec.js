import React from 'react'
import { getBlankQuest, getQuestFromProps, getQuizStatefromQuest } from '../quizHelper'

describe('quizHelper', () => {

  it('getBlankQuest should return blank', () => {
    expect(getBlankQuest()).toMatchSnapshot()
  })

  it('getQuestFromProps should return blank if quest is undefined', () => {
    expect(getQuestFromProps()).toMatchSnapshot()
  })

  it('getQuestFromProps should return new object', () => {
    const quest = {
      subject: 'subject',
      question: 'question',
      choices: ['c0', 'c1', 'c2']
    }
    expect(getQuestFromProps(quest)).toMatchSnapshot()
  })

  it('getQuizStatefromQuest should return state with blank if quest is new object', () => {
    expect(getQuizStatefromQuest({})).toMatchSnapshot()
  })

  it('getQuizStatefromQuest should return state with data', () => {
    const quest = {
      subject: 'subject',
      question: 'question',
      choice_0: 'c0',
      choice_1: 'c1',
      choice_2: 'c2',
    }
    expect(getQuizStatefromQuest(quest)).toMatchSnapshot()
  })
})