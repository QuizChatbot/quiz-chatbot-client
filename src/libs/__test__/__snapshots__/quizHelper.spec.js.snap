// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quizHelper getBlankQuest should return blank 1`] = `
Object {
  "choice_0": "",
  "choice_1": "",
  "choice_2": "",
  "question": "",
  "subject": "",
}
`;

exports[`quizHelper getQuestFromProps should return blank if quest is undefined 1`] = `Object {}`;

exports[`quizHelper getQuestFromProps should return new object 1`] = `
Object {
  "choice_0": "c0",
  "choice_1": "c1",
  "choice_2": "c2",
  "question": "question",
  "subject": "subject",
}
`;

exports[`quizHelper getQuizStatefromQuest should return state with blank if quest is new object 1`] = `
Object {
  "choice_0": "",
  "choice_1": "",
  "choice_2": "",
  "isEditing": Object {
    "choice_0": false,
    "choice_1": false,
    "choice_2": false,
    "question": false,
    "subject": false,
  },
  "question": "",
  "subject": "",
}
`;

exports[`quizHelper getQuizStatefromQuest should return state with data 1`] = `
Object {
  "choice_0": "c0",
  "choice_1": "c1",
  "choice_2": "c2",
  "isEditing": Object {
    "choice_0": false,
    "choice_1": false,
    "choice_2": false,
    "question": false,
    "subject": false,
  },
  "question": "question",
  "subject": "subject",
}
`;
